src/base/bittorrent/session.cpp

```diff
@@ -4127,8 +4122,20 @@ quint64 Session::getAlltimeUL() const
     return m_statistics->getAlltimeUL();
 }
 
+void Session::initiateSessionRefresh()
+{
+    if (!m_sessionRefreshed) {
+        // we have to wait for two alerts, calling of this function means one of them is completed
+        // this means we can safely enqueue new refresh request in next call
+        m_sessionRefreshed = true;
+        return;
+    }
+    QTimer::singleShot(refreshInterval(), this, &Session::refresh);
```

Comment:
I still think my previous proposal is better approach to keep real refresh interval closest to setting value.

Reply:
have to pass around intervals and all, I think a loss of 1 millisecond is inconsequential. so I didn't implement this